opentelemetry-appender-log/src/lib.rs

```diff
@@ -31,7 +31,7 @@ where
             self.logger.emit(
                 LogRecordBuilder::new()
                     .with_severity_number(map_severity_to_otel_severity(record.level()))
-                    .with_observed_timestamp(SystemTime::now())
+                    .with_timestamp(SystemTime::now())
```

Comment:
I feel the `timestamp` should be populated by the logging framework, which is `tokio-tracing` here. And not the appender. It should remain empty if the logging framework is not generating it.

Reply:
tokio-tracing itself does not populate it as it is merely a "facade", outsourcing most of the work to "subscribers". In this case, the subscriber *is* the also the log appender, so I think this is the right place to fetch timestamp.

Reply:
https://github.com/tokio-rs/tracing-opentelemetry/blob/v0.1.x/src/layer.rs#L842 current tracing-opentelemetry impln. (its not for logs, but just sharing to show tracing crate itself is not in charge of timestamps)

Reply:
Yeah it's tricky. Log Appender shouldn't populate timestamp (as per otel specs), but subscriber could.
And opentelemetry-appender-log crate is both. I was thinking of scenario where the tracing subscriber is chained with multiple layers, and one of the intermediate layer is enriching the timestamp which appender can use. But probably not that frequent use-case.

But I don't have strong opinion here, should be good.

Reply:
Yes! 
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Logs/ILogger/OpenTelemetryLogger.cs#L91 Doing this in the OpenTelemetry bridge/appender is the only way ! This is from .NET appender. Its possible there are multiple subscribers for .NET as well, but everyone is responsible to set own timestamp!

Reply:
Btw, Log spec does not forbid populating timestamp, so we good from that perspective.

Reply:
This seems reasonable for now, we can get feedback on the design as people start integrating it as well